plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -311,6 +318,15 @@ func (k *KafkaConsumer) Start(acc telegraf.Accumulator) error {
 			handler := NewConsumerGroupHandler(acc, k.MaxUndeliveredMessages, k.parserFunc, k.Log)
 			handler.MaxMessageLen = k.MaxMessageLen
 			handler.TopicTag = k.TopicTag
+			//if message headers list specified, put it as map to handler
+			MsgHeadersMap := make(map[string]bool)
```

Comment:
Should we preallocate the size of the map?

```suggestion
			MsgHeadersMap := make(map[string]bool, len(k.MsgHeadersAsTags))
```

Reply:
Won't it fail when headers are not specified?

Reply:
`MsgHeadersAstags` will return the default value which is an empty []string. Then the length will be 0.